modules/login/php/module.class.inc

```diff
@@ -3,35 +3,67 @@
  * This serves as a hint to LORIS that this module is a real module.
  * It does nothing but implement the module class in the module's namespace.
  *
- * PHP Version 5
+ * PHP Version 7
  *
- * @category   Core
- * @package    Main
- * @subpackage Login
- * @author     Dave MacFarlane <david.macfarlane2@mcgill.ca>
- * @license    http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
- * @link       https://www.github.com/aces/Loris/
+ * @category LORIS_Module
+ * @package  Login
+ * @author   Alizée Wickenheiser <alizee.wickenheiser@mcin.ca>
+ * @license  http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link     https://www.github.com/aces/Loris-Trunk/
  */
 namespace LORIS\login;
 
+use Psr\Http\Message\ResponseInterface;
+use Psr\Http\Message\ServerRequestInterface;
+
 /**
  * Class module implements the basic LORIS module functionality
  *
- * @category   Core
- * @package    Main
- * @subpackage Login
- * @author     Dave MacFarlane <david.macfarlane2@mcgill.ca>
- * @license    http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
- * @link       https://www.github.com/aces/Loris/
+ * @category LORIS_Module
+ * @package  Login
+ * @author   Alizée Wickenheiser <alizee.wickenheiser@mcin.ca>
+ * @license  http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link     https://www.github.com/aces/Loris-Trunk/
  */
 class Module extends \Module
 {
+    /**
+     * The default handler for a module routes to a page of the first component of
+     * the URL's name, or the same name as the module if serving the root of the
+     * module.
+     *
+     * Modules extend PrefixRouter, so that the parent handler knows that the
+     * css, static, and js routes should be routed through the file system to
+     * retrieve the files before trying to load any specific pages.
+     *
+     * @param ServerRequestInterface $request The incoming PSR7 request
+     *
+     * @return ResponseInterface the outgoing PSR7 response
+     */
+    public function handle(ServerRequestInterface $request) : ResponseInterface
+    {
+        $path = trim($request->getURI()->getPath(), '/');
+        switch ($path) {
+        case 'Authentication':
+            \Module::factory('api');
+            $handler = new endpoints\Authentication();
+            break;
+        default:
+            return parent::handle($request);
+        }
+        $user = $request->getAttribute('user');
+        if (!$handler->_hasAccess($user)) {
+            return new \LORIS\Http\Response\JSON\Forbidden();
+        }
```

Comment:
I think You don't need this and instead this module should override the parent class's `isPublicModule()` function to be true. Every user should be able to access this module.

Reply:
@johnsaigle I'm going to do similar to this PR from the comments received from @driusan about the issue_tracker PR.